platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+The `config.txt` will never be used again in the network life cycle.
+
+Given that:
+
+1. we need a special mechanism for supplying a genesis roster to a brand new network, and
+2. this mechanism is only ever used once during genesis, and
+3. we already have a support for `config.txt` in the platform, and
+4. we already propose a public interface for Services to install Candidate Rosters in existing networks
+
+there should be absolutely no need to introduce any additional mechanisms or interfaces between Platform and Services to
+supply a genesis roster.
+
+If we must, we may add a field to the `config.txt` to indicate the number of shares for each node.
+This should be a relatively trivial change.
+However, since the `config.txt` would only ever be used once during genesis, it’s a lot simpler to skip that and assume
+equal shares for all genesis nodes, and only start supporting non-equal shares when a new Candidate Roster is installed
+via the new public interface of the Platform API. But as mentioned, if we must support this at genesis for brand new
+networks, we can do that, and it should be trivial.
+
+### Roster changes needed for Components
+
+- Reconnect. Reconnect logic currently exchanges and validates address book between the learner and teacher nodes. The
+  learner node uses the address book to select the teacher node, as well as remove all invalid signatures from the
+  state. Both of these will need to be updated to use rosters instead.
+- Networking. Most parts of the network code have been abstracted away from the address book as part of the Dynamic
+  Address Book effort, so there should be minimal work left there.
+- Event Validation. The event validation logic uses the address book to determine whether a given event has a valid
+  signature. This will be updated to use rosters instead.
+- Miscellaneous components. Other components that reference things stored in the address book (like consensus needing
+  the node weight or the crypto module verifying keys) will need to be updated to use things stored in rosters instead.
+
+### Test Plan
+
+Some of the obvious test cases to be covered in the plan include validating one or more of the following scenarios:
+
+1. New valid Candidate Roster created with no subsequent one sent by app. Verify accept.
+2. New valid Candidate Roster created with subsequent one sent by app. Verify accept.
+3. Invalid roster(s) sent. Verify reject.
+4. Empty roster(s) sent. Verify reject.
+5. Node Failures During Roster Change: What happens if nodes fails or disconnects during a roster change? Verify valid
+   node successfully reconnects.
+6. Concurrent Roster Updates: What if we make multiple roster updates concurrently? Verify no effect on adoption.
+7. Rosters with nodes with zero or maximum shares. What happens?
+8. Roster recovery? Node receives roster, crashes. Wakes up … ?
+9. What testing do we need for genesis new network?
```

Comment:
The answer to this question is?

Reply:
I don't know actually. One approach though is to treat the platform as a library, in which case, we need not do anything specific to testing a genesis new network - a network is a network, we just test that.

Reply:
removed & resolved